tests/Moq.Tests/MockFixture.cs

```diff
@@ -1312,5 +1312,37 @@ public void Accessing_property_of_bad_serializable_type_after_SetupAllProperties
 			Assert.ThrowsAny<Exception>(() => mock.Object.BadSerializable);
 		}
 #endif
+
+		public interface ITestInterface
+		{
+			void Get(Guid id);
+		}
+
+		[Fact]
+		public void Thousand_different_setups_can_be_done_in_under_a_second()
+		{
+			var sw = System.Diagnostics.Stopwatch.StartNew();
+			var mock = new Mock<ITestInterface>();
+			for (int i = 0; i < 1000; i++)
+			{
+				mock.Setup(m => m.Get(Guid.NewGuid()));
+			}
+			sw.Stop();
+			Assert.True(sw.Elapsed < TimeSpan.FromSeconds(1), $"Setups were expected to take less than a second but it took {sw.Elapsed}");
+		}
+
+		[Fact]
+		public void Thousand_same_setups_can_be_done_in_under_a_second()
+		{
+			var sw = System.Diagnostics.Stopwatch.StartNew();
+			var mock = new Mock<ITestInterface>();
+			var id = Guid.NewGuid();
+			for (int i = 0; i < 1000; i++)
+			{
+				mock.Setup(m => m.Get(id));
+			}
+			sw.Stop();
+			Assert.True(sw.Elapsed < TimeSpan.FromSeconds(1), $"Setups were expected to take less than a second but it took {sw.Elapsed}");
+		}
```

Comment:
Timed unit tests are a bad idea because we don't really have a way of determining an absolute time threshold. These tests may never fail to begin with on a really fast machine; and they may still fail (even after your proposed change) on a very slow machine.

Perhaps the only reference time we have is that of previous Moq versions... but those cannot be easily used in this test suite.

I can't think of a good way to write these unit tests, so unless you can, I suggest we remove them.

Reply:
i fully agree with timed unittests being a bad idea but i didn't know a different way of testing the problem but i'm fine with removing them